### ISSUE TITLE
Additional variants for people to use

### ISSUE THREAD

**Issue by StellaAthena** [r-three/common-pile#130]
I've gotten a lot of feedback from people about how the form that the data was released in doesn't really meet their needs. Sometimes this is because they want something that will work out of the box and that they don't need to think about, sometimes this is because they want to study the Comma v0.1 models and aren't sure exactly how to transform the data we released into the data that the model saw. The requests I have seen so far are for:
1. The plain text of the data we trained on, up/down sampled appropriately.
2. The plain text of the data we trained on, up/down sampled appropriately, in the right order and labeled by gradient update step (though not tokenized or broken into sequences).
3. The exact data we trained on – tokenized, shuffled, upsampled, and packaged into sequences – and labeled by gradient update step.

People have also expressed interest in representative smaller slices of the Common Pile. In addition to releasing such things it would also be a good idea to release a function that takes a number of tokens as an input and produces a slice of that size as an output.

Another really common question is whether we have a held-out validation set.

How much of this do we already have from the training process? I can make a guess at how to produce this using Lingua, but I've never used the library before and wouldn't be able to validate that I did it correctly without access to the actually-trained-on text anyways.

@craffel

**Comment by craffel** [comment]
Thanks for flagging these requests.

2 and 3 would be reconstructable via lingua, and so would the function, would just take some effort that I currently don't have the bandwidth for. I'm not sure what is meant by 1 (i.e., without caring about the order), but it seems like you could get that from 2 (since if order doesn't matter, then the order imposed by 2 can just be ignored if you want 1). Of course we'd have to do all of this separately for the 1T and 2T variants.

Generating a representative slice would of course be very straightforward, no lingua required.

No, there's no held-out validation set and I personally don't see value in producing validation sets for this kind of data since, at this time, it's not being proposed for any sort of benchmarking (and using a validation set for seeing how training is going can be done ad-hoc/individually). If someone wanted to propose a common pile-based benchmark then I could see specifying splits explicitly.

**Comment by StellaAthena** [comment]
> 2 and 3 would be reconstructable via lingua, and so would the function, would just take some effort that I currently don't have the bandwidth for.

I'm down to try to do it if you share the config / options used when the training was launched and am worried about validating it. Do we have a copy of the processed data used for training saved anywhere that I can compare to?

> I'm not sure what is meant by 1 (i.e., without caring about the order), but it seems like you could get that from 2 (since if order doesn't matter, then the order imposed by 2 can just be ignored if you want 1).

Good point!

> Of course we'd have to do all of this separately for the 1T and 2T variants.

Yup

> No, there's no held-out validation set and I personally don't see value in producing validation sets for this kind of data since, at this time, it's not being proposed for any sort of benchmarking (and using a validation set for seeing how training is going can be done ad-hoc/individually). If someone wanted to propose a common pile-based benchmark then I could see specifying splits explicitly.

It's common in interpretability and when studying memorization to want to have something IID from the training data distribution but not part of the training data.
